PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs

```diff
@@ -348,12 +421,27 @@ public Collection<Variable> GetScopedVariable()
 
             foreach (var psobj in _varaiables)
             {
-                var psVar = psobj.BaseObject as PSVariable;
+                if (_runspace.ConnectionInfo == null)
+                {
+                    var psVar = psobj.BaseObject as PSVariable;
 
-                if (psVar != null)
+                    if (psVar != null)
+                    {
+                        _localVariables.Add(psVar);
+                        variables.Add(new Variable(psVar));
+                    }
+                }
+                else
                 {
-                    _localVariables.Add(psVar);
-                    variables.Add(new Variable(psVar));
+                    dynamic psVar = (dynamic)psobj;

```

Comment:
For my knowledge, what types do we expect here?  PSObject/primitives/IEnumerables/everything else?